Sources/WeaverCodeGen/SwiftGenerator.swift

```diff
@@ -725,12 +725,27 @@ static func _pushDynamicResolver<Resolver>(_ resolver: Resolver) {
                     guard selfReferenceDeclarations.contains(declaration) == false else { return nil }
 
                     let assignment: (MetaDependencyDeclaration) -> Assignment = { resolvedDeclaration in
-                        Assignment(
-                            variable: Variable._self.reference + .named(declaration.buildersSubcriptGet),
-                            value: Variable._self.reference + .named("builder") | .call(Tuple()
-                                .adding(parameter: TupleParameter(value: Reference.named(resolvedDeclaration.declarationName)))
+                        switch reference.kind {
+                        case .parameter:
+                            return Assignment(
+                                variable: Variable._self.reference + .named(declaration.buildersSubcriptGet),
+                                value: Variable._self.reference + .named("builder") | .call(Tuple()
+                                    .adding(parameter: TupleParameter(value: Reference.named(resolvedDeclaration.declarationName)))
+                                )
                             )
-                        )
+
+                        case .reference:
+                            return Assignment(
+                                variable: Variable._self.reference + .named(declaration.buildersSubcriptGet),
+                                value: .named("getBuilder") | .call(Tuple()
+                                    .adding(parameter: TupleParameter(name: "for", value: Value.string(resolvedDeclaration.declarationName)))
+                                    .adding(parameter: TupleParameter(name: "type", value: resolvedDeclaration.type.typeID.reference + .named(.`self`)))
+                                )
+                            )
+
+                        case .registration:
+                            fatalError("Invalid kind for input reference")
```

Comment:
Please add a case to `SwiftGeneratorError` and throw it here instead of using `fatalError`.

Also, I'm not sure that `Invalid kind for input reference` is detailed enough for developers to understand what's happening.

Reply:
Hmm, it's some kind of logical error, the reference cannot be of `registration` kind the same time it is an input reference. I tried my best and named this error as `inputReferenceCannotBeOfRegistrationKind` but I'll be grateful if you would give it a more suitable name :)